objects/src/accounts/delta/storage.rs

```diff
@@ -25,6 +25,29 @@ pub struct AccountStorageDelta {
 }
 
 impl AccountStorageDelta {
+    /// Creates an empty [AccountStorageDelta].
+    pub fn empty() -> Self {
+        Self {
+            cleared_items: Vec::new(),
+            updated_items: Vec::new(),
+            updated_maps: Vec::new(),
+        }
+    }
+
+    /// Creates an [AccountStorageDelta] from the given iterators.
+    pub fn from_iterators<A, B, C>(cleared_items: A, updated_items: B, updated_maps: C) -> Self
+    where
+        A: IntoIterator<Item = u8>,
+        B: IntoIterator<Item = (u8, Word)>,
+        C: IntoIterator<Item = (u8, StorageMapDelta)>,
+    {
+        Self {
+            cleared_items: Vec::from_iter(cleared_items),
+            updated_items: Vec::from_iter(updated_items),
+            updated_maps: Vec::from_iter(updated_maps),
+        }
+    }
```

Comment:
I think these are good but I didn't see a point where iterators would be preferred over just passing the `Vec` in the code. 

With this and the previous comment in mind, I wonder if the API should be `AccountStorageDelta::new(Vec, Vec, Vec)`, `AccountStorageDelta::default()` for empty lists and then `AccountStorageDelta::from_iterators(IntoIter, IntoIter, IntoIter)` if needed (ie, remove it if not needed although there might be a usecase I'm not keeping in mind or even future users of this API, etc.)

Reply:
> I think these are good but I didn't see a point where iterators would be preferred over just passing the Vec in the code.

The idea here is to make the code [more reusable](https://rust-lang.github.io/api-guidelines/flexibility.html#functions-minimize-assumptions-about-parameters-by-using-generics-c-generic). So this version supports both `vec`s and `slice`s.

> With this and the previous comment in mind, I wonder if the API should be AccountStorageDelta::new(Vec, Vec, Vec), AccountStorageDelta::default() for empty lists and then AccountStorageDelta::from_iterators(IntoIter, IntoIter, IntoIter) if needed (ie, remove it if not needed although there might be a usecase I'm not keeping in mind or even future users of this API, etc.)

In terms of API design. I think `from_iterators` is pretty bad, because it is basically unclear what the are the values of each iterator used for. My personal preference is to use a builder pattern, e.g. `with_cleared_items` / `with_updated_items` / `with_updated_maps`. In general, any time there are two arguments of the same type, `bool, bool`, `Option<Vec<_>>`, etc, I would prefer a builder pattern to make the code easier to follow. However, I have made similar changes in the past they have been rejected in PR review, so here I went for a single method.

Regarding your suggestion, I don't have strong preferences, but the `from_iterators` is a bit more generic, I can add the `new` if you want.

Reply:
For now I renamed the `from_iterators` to `from_iters`, I guess the second is more inline with our codebase. And I marked the interfaces private, so we can change them in the future with no worries.